mscore/startupWizard.cpp

```diff
@@ -0,0 +1,105 @@
+//=============================================================================
+
+#include "startupWizard.h"
+#include "musescore.h"
+#include "libmscore/mscore.h"
+#include "shortcut.h"
+
+namespace Ms {
+
+const static QMap<QString, QString> layoutToShortcut = {
+      {QT_TRANSLATE_NOOP("keyboard-layout", "US-QWERTY"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "UK-QWERTY"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "QWERTZ"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "AZERTY"), ":/data/shortcuts_AZERTY.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "BEPO"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "DVORAK"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "JIS - Standard Japanese"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "OYAYUBI SHIFUTO"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "T1 - Standard German"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "T2 - German"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Swiss German"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Belgian French"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Canadian French"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "French French"), ":/data/shortcuts_AZERTY.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Swiss French"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Spanish"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Italian"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Brazilian Portuguese ABNT"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Portuguese"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "US - International"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Dutch"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Korean - Dubeolsik"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Korean - Sebeolsik"), ":/data/shortcuts.xml"},
+      {QT_TRANSLATE_NOOP("keyboard-layout", "Other"), ":/data/shortcuts.xml"},
+      };
+
+const static QMap<QString, QString> langToLayout = {
+      {"en", "US-QWERTY"},
+      {"ja", "JIS - Standard Japanese"},
+      {"de", "T1 - Standard German"},
+      {"fr", "French French"},
+      {"es", "Spanish"},
+      {"it", "Italian"},
+      {"pt", "Portuguese"},
+      {"nl", "Dutch"},
+      {"ko", "Korean - Dubeolsik"}
+      };
+
+StartupWizardPage1::StartupWizardPage1(QWidget* parent)
+    : QWizardPage(parent)
+      {
+      setTitle(QObject::tr("Keyboard Layout"));
+      QLabel *label = new QLabel(QObject::tr("Enter your keyboard layout"));
```

Comment:
same, here and for the combo, add this as the parent for the QLabel and the combo, the layout. Then they will be deleted when the startupwizard is deleted. If not, you have a memory leak.
Read http://doc.qt.io/qt-5/objecttrees.html again